ligo/haskell/src/Ligo/Typescript.hs

```diff
@@ -0,0 +1,448 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+module Ligo.Typescript (generateTs) where
+
+import Universum
+
+import Data.Char (toUpper)
+import qualified Data.Text as T
+import Fmt (Buildable(build), blockListF', (+|), (|+))
+import qualified Fmt
+import System.FilePath
+
+import Lorentz.Entrypoints
+import Michelson.Typed.Annotation
+import Michelson.Typed.Entrypoints (pnNotes)
+import Michelson.Untyped.Annotation
+
+-- | This is mostly same as the 'Notes t' type, but without the type
+-- parameter and without annotations that we are not interested.
+data EpType
```

Comment:
There seem to be other partial re-implementation of some `morley` types and functions here.

Have you tried re-using the existing ones instead? If so where there any particular issues?

I'm a bit concerned with maintaining this, especially considering that (very cool BTW) such approach allows for re-generation and it would be a bonus to have it auto-update for the most part.

Reply:
>If so where there any particular issues?

I tried to use `mkEntrypointsMap` from morley that can convert an untyped `Type` to a map of entrypoint -> untyped types. Here we had a typed `Type` and wanted to make entrypoint names out of it. The issue here is that the typed `Type` cannot hold annotations, so they are dervived from the actual type via another typeclass, which appear to be what `parameterEntrypointsToNotes` function is doing. 

We would be able to use the `mkEntrypointsMap` function if we can make a untyped type from a `Notes t` stucture, without going through typed T first, because that would lose all annotations. 

I should be possible to get rid of `EpType` and some other types. ll try to do it.

>I'm a bit concerned with maintaining this

I am not sure if this can be generally implemented for all parameters, but if it is, may be it is worth extracting it to a separate package or include it in an existing one. What do you think?

Reply:
I have refactored the code to get rid of EpTypes and also ended re-using `mkEntrypointsMap` function from `morley`.